nameserver/server.go

```diff
@@ -32,7 +20,94 @@ func makeDNSFailResponse(r *dns.Msg) *dns.Msg {
 	return m
 }
 
-func queryHandler(lookups []Lookup) dns.HandlerFunc {
+type DNSServer struct {
+	config *dns.ClientConfig
+	zone   Zone
+	iface  *net.Interface
+	port   int
+	udpSrv *dns.Server
+	tcpSrv *dns.Server
+}
+
+// Creates a new DNS server with a given config
+func NewDNSServerWithConfig(config *dns.ClientConfig, zone Zone, iface *net.Interface, port int) (*DNSServer, error) {
+	return &DNSServer{
+		config: config,
+		zone:   zone,
+		iface:  iface,
+		port:   port,
+	}, nil
+}
+
+// Creates a new DNS server
+func NewDNSServer(zone Zone, iface *net.Interface, port int) (*DNSServer, error) {
```

Comment:
There are two usage cases in this PR: we create a concrete `dns.ClientConfig` and the invoke `NewDNSServerWithConfig` in unit tests, but the common case is to use `NewDNSServer`. Passing a _low-level_ parameter like `dns.ClientConfig` by default is not very elegant/useful, so that is the reason why `DNSServer` is properly initialized with a [DNSServerConfig](https://github.com/zettio/weave/pull/394/files#diff-7d7cca87c8b1140efab81f1f58b9c0edR25) in the [next PR](https://github.com/zettio/weave/pull/394). I can change it here anyway...